### PR TITLE
[Az.Resources Policy] Add -NonComplianceMessage to Set/New-AzPolicyAssignment

### PR DESCRIPTION
**Pull request by pilor**
## Description

Adding a new parameter (-NonComplianceMessage) to Set-AzPolicyAssignment and New-AzPolicyAssignment. This parameter contains an array of non compliance message definitions that define the messages users will get if their resources are denied by policy.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/942
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Resources/Resources/help/New-AzPolicyAssignment.md [hunk 1]
```diff
@@ -152,6 +153,17 @@ The second command gets the policy definition named VirtualMachinePolicy by usin
...
 The assignment is set with an EnforcementMode value of _DoNotEnforce_ i.e. the policy effect is not enforced during resource creation or update.
 
+### Example 7: Policy assignment with non-compliance messages
+```
+PS C:\> $PolicySet = Get-AzPolicySetDefinition -Name 'VirtualMachinePolicySet'
+PS C:\> $NonComplianceMessages = @(@{Message="Only DsV2 SKUs are allowed."; PolicyDefinitionReferenceId="DefRef1"}, @{Message="Virtual machines must follow cost management best practices."})
+PS C:\> New-AzPolicyAssignment -Name 'VirtualMachinePolicyAssignment' -PolicySetDefinition $PolicySet -NonComplianceMessage $NonComplianceMessages
+```
+
+The first command gets the policy set definition named VirtualMachinePolicy by using the Get-AzPolicySetDefinition cmdlet and stores it in the $PolicySet variable.
```
Comment (mentat9) [inline src/Resources/Resources/help/New-AzPolicyAssignment.md:163]: VirtualMachinePolicySet

**Comment by mentat9** [comment]
Since you are doing this change already, it would be great if you could also remove the two CmdletParameterBreakingChange attributes from New-AzPolicyAssignment as well, since that change shipped long ago.

**Comment by mentat9** [comment]
>         [Parameter(Mandatory = false, ValueFromPipeline = true, ValueFromPipelineByPropertyName = true, HelpMessage = PolicyHelpStrings.NewPolicyAssignmentPolicyDefinitionHelp)]

This can be removed: the change shipped long ago. Same with the other one on the next parameter.

---
Refers to: src/Resources/ResourceManager/Implementation/Policy/NewAzurePolicyAssignment.cs:79 in 2a90471. [](commit_id = 2a904716fb9893f303fdc96a9150cede6f1fc1ff, deletion_comment = False)

**Comment by mentat9** [comment]
> 

Since you are here, could you add a note that RBAC needs to be added separately? Something like appending "Note: the newly created identity must be manually configured with the correct role assignments."

---
Refers to: src/Resources/Resources/help/New-AzPolicyAssignment.md:187 in 2a90471. [](commit_id = 2a904716fb9893f303fdc96a9150cede6f1fc1ff, deletion_comment = False)

**Review by mentat9 — DISMISSED** [review]
See comments.

**Comment by msJinLei** [comment]
@mentat9 are you going to approve this PR?

**Review by mentat9 — APPROVED** [review]

**Review by msJinLei — APPROVED** [review]

**Pull request merged**